cypress/e2e/shared/checks.test.ts

```diff
@@ -115,6 +116,129 @@ describe('Checks', () => {
+        expect(resp.name).to.equal(thresholdCheck.name)
+      })
+      cy.reload()
+      cy.intercept('PUT', '**/checks/*').as('putCheck')
+      cy.getByTestID('check-card--name')
+        .eq(0)
+        .click()
+      cy.getByTestID('overlay').should('be.visible')
+      // Properties
+      // Scheduler properties
+      cy.getByTestID('schedule-check').click()
+      cy.getByTestID('dropdown-menu--contents').within(() => {
+        cy.contains('5m').click()
+      })
+      cy.getByTestID('offset-options').click()
+      cy.getByTestID('dropdown-menu--contents').within(() => {
+        cy.contains('1m').click()
+      })
+      // Tags
+
+      for (let i = 0; i < testTags.length; i++) {
+        cy.getByTestID('dashed-button').click()
+        cy.getByTestID('tag-rule-key--input')
+          .eq(i)
+          .type(testTags[i].key)
+        cy.getByTestID('tag-rule-value--input')
+          .eq(i)
+          .type(testTags[i].value)
+      }
+
+      // Status Message Template
+
+      cy.getByTestID('status-message-textarea')
+        .clear()
+        .type(newMsg, {parseSpecialCharSequences: false})
+
+      // Thresholds
+      // remove one threshold
+      cy.getByTestID('panel-OK').within(() => {
+        cy.getByTestID('add-threshold-condition-OK').should('not.exist')
+        cy.getByTestID('dismiss-button').click()
+        cy.wait(500)
+      })
+      cy.getByTestID('add-threshold-condition-OK').should('be.visible')
+
+      // change logic of one threshold
+      cy.getByTestID('panel-INFO').within(() => {
+        cy.getByTestID('dropdown--button')
+          .should('contain.text', 'is above')
+          .click()
+        cy.getByTestID('dropdown-menu--contents').within(() => {
+          cy.contains('is inside range').click()
+        })
+        cy.getByTestID('input-field')
+          .should('have.length', 2)
+          .eq(0)
+          .clear()
+          // seem to be encountering cypress issue here https://github.com/cypress-io/cypress/issues/3817
+          .type(` ${rangeBtm}{del}`) // workaround
+        cy.getByTestID('input-field')
+          .eq(0)
+          .should('have.value', rangeBtm)
+        cy.getByTestID('input-field')
+          .eq(1)
+          .clear()
+          // seem to be encountering cypress issue here https://github.com/cypress-io/cypress/issues/3817
+          .type(` ${rangeTop}{del}`) // workaround
+        cy.getByTestID('input-field')
+          .eq(1)
+          .should('have.value', rangeTop)
+      })
+      // change value of one threshold
+      cy.getByTestID('panel-WARN').within(() => {
+        cy.getByTestID('input-field-WARN')
+          .clear()
+          // seem to be encountering cypress issue here https://github.com/cypress-io/cypress/issues/3817
+          .type(` ${rangeTop}{del}`) // workaround
+      })
+      cy.getByTestID('save-cell--button').click()
+      cy.wait('@putCheck').then(({response}) => {
+        if (response) {
```

Comment:
Suggestion: To reduce cyclomatic complexity, you can just do this:
```
if (!response) {
    fail('No response after update check')
    return
}

expect(response.body.tags).to.deep.equal(testTags)
expect(response.body.every).to.equal(every)
expect(response.body.offset).to.equal(offset)
expect(response.body.statusMessageTemplate).to.equal(newMsg)
expect(
  response.body.thresholds.filter((th: any) => th.level === 'INFO')[0]
    .min
).to.equal(parseFloat(rangeBtm))
expect(
  response.body.thresholds.filter((th: any) => th.level === 'INFO')[0]
    .max
).to.equal(parseFloat(rangeTop))
expect(
  response.body.thresholds.filter((th: any) => th.level === 'WARN')[0]
    .value
).to.equal(parseFloat(rangeTop))
```

Reply:
Refactored as per suggestion